client/src/utils/api-util.js

```diff
@@ -95,6 +95,38 @@ export async function postFile(url, file) {
     else return res;
 }
 
+/**
+ * Send a PUT request to the specified url with the specified data
+ * @param url - URL to send the PUT to
+ * @param data - JSON data to send to the specified URL
+ */
+export async function put(url, data) {
```

Comment:
You should abstract the code from `POST` into a generic function that accepts the method type, no need to copy and paste. Also there will be conflicts with my PR because I made content type configurable